plugins/shell/src/process/mod.rs

```diff
@@ -84,6 +84,12 @@ impl CommandChild {
     pub fn pid(&self) -> u32 {
         self.inner.id()
     }
+
+    /// Waits for the child to exit completely, returning the status that it exited with.
+    pub fn wait(self) -> crate::Result<std::process::ExitStatus> {
```

Comment:
It will block the whole app to wait for the process to exit, we need to use something async / non blocking

Reply:
I use async fn

Reply:
async fn will also block it